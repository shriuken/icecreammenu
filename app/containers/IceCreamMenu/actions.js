/*
 *
 * IceCreamMenu actions
 *
 */

import {
  DEFAULT_ACTION,
  ADD_FLAVOR,
  DELETE_FLAVOR,
  EDIT_FLAVOR,
} from './constants';

export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}

export function addFlavor(flavor) {
  return {
    type: ADD_FLAVOR,
    flavor,
  };
}

export function deleteFlavor(flavorId) {
  // console.log('deleteFlavor');
  return {
    type: DELETE_FLAVOR,
    flavorId,
  };
}

export function editFlavor(flavor) {
  return {
    type: EDIT_FLAVOR,
    flavor,
  };
}
