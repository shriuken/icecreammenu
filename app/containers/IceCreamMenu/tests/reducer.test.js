import { fromJS } from 'immutable';
import iceCreamMenuReducer from '../reducer';
import { addFlavor, deleteFlavor /* editFlavor */ } from '../actions';

import data from '../data-object.json';

const immData = fromJS(data);
const filteredData = immData.set(
  'flavorsById',
  immData.get('flavorsById').filter(flavor => flavor.get('active')),
);

const flavor = fromJS({
  id: 'uniqueStringThatsNotInDatabaseHopefully',
  name: "I can't believe it's not cinammon",
  price: 4.0,
  ingredients: 'cocoa powder, milk, sugar, salt',
  active: true,
});

describe('iceCreamMenuReducer', () => {
  it('uses mockdata', () => {
    expect(iceCreamMenuReducer(undefined, {})).toEqual(filteredData);
  });
  it('returns the initial state', () => {
    expect(iceCreamMenuReducer(undefined, {})).toMatchSnapshot();
  });
  it('handles the ADD_FLAVOR action', () => {
    const state = iceCreamMenuReducer(undefined, addFlavor(flavor));
    expect(state).toMatchSnapshot();
    expect(state.getIn(['flavorsById', flavor.get('id')])).not.toEqual(
      undefined,
    );
  });
  it('handles the delete flavor action', () => {
    const state = iceCreamMenuReducer(undefined, deleteFlavor('0'));
    expect(state).toMatchSnapshot();
    expect(state.getIn(['flavorsById', '0'])).toEqual(undefined);
  });

  // test edit flavor reducer here too.
});
