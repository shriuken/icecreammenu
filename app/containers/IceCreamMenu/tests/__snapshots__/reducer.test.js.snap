// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iceCreamMenuReducer handles the ADD_FLAVOR action 1`] = `
Immutable.Map {
  "flavorsById": Immutable.Map {
    "0": Immutable.Map {
      "id": "0",
      "name": "vanilla",
      "price": 4,
      "ingredients": "vanilla bean, milk, sugar, cream",
      "active": true,
    },
    "1": Immutable.Map {
      "id": "1",
      "name": "chocolate",
      "price": 4,
      "ingredients": "cocoa powder, milk, sugar, salt",
      "active": true,
    },
    "uniqueStringThatsNotInDatabaseHopefully": Immutable.Map {
      "id": "uniqueStringThatsNotInDatabaseHopefully",
      "name": "I can't believe it's not cinammon",
      "price": 4,
      "ingredients": "cocoa powder, milk, sugar, salt",
      "active": true,
    },
  },
}
`;

exports[`iceCreamMenuReducer handles the delete flavor action 1`] = `
Immutable.Map {
  "flavorsById": Immutable.Map {
    "1": Immutable.Map {
      "id": "1",
      "name": "chocolate",
      "price": 4,
      "ingredients": "cocoa powder, milk, sugar, salt",
      "active": true,
    },
  },
}
`;

exports[`iceCreamMenuReducer returns the initial state 1`] = `
Immutable.Map {
  "flavorsById": Immutable.Map {
    "0": Immutable.Map {
      "id": "0",
      "name": "vanilla",
      "price": 4,
      "ingredients": "vanilla bean, milk, sugar, cream",
      "active": true,
    },
    "1": Immutable.Map {
      "id": "1",
      "name": "chocolate",
      "price": 4,
      "ingredients": "cocoa powder, milk, sugar, salt",
      "active": true,
    },
  },
}
`;
